pkg/configurator/crd_client_test.go

```diff
@@ -0,0 +1,257 @@
+package configurator
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	tassert "github.com/stretchr/testify/assert"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	testclient "github.com/openservicemesh/osm/pkg/gen/client/config/clientset/versioned/fake"
+
+	"github.com/openservicemesh/osm/pkg/announcements"
+	"github.com/openservicemesh/osm/pkg/apis/config/v1alpha1"
+	"github.com/openservicemesh/osm/pkg/kubernetes/events"
+)
+
+const (
+	osmMeshConfigName = "-test-osm-mesh-config-"
+)
+
+var _ = Describe("Test OSM MeshConfig parsing", func() {
+	defer GinkgoRecover()
+
+	meshConfigClientSet := testclient.NewSimpleClientset()
+
+	stop := make(<-chan struct{})
```

Comment:
I suggest you initialize first pubsub channel, and after the configurtor, to make sure subscription has been processed by pubsub lib

Reply:
context:
https://github.com/openservicemesh/osm/issues/3144